py/desitarget/mtl.py

```diff
@@ -138,7 +138,16 @@ def make_mtl(targets, obscon, zcat=None, trim=False, scnd=None):
         if (obsconditions.mask(obscon) & obsconditions.mask("BRIGHT")) != 0:
             ii = targets_zmatcher[desi_target] & desi_mask.BGS_ANY > 0
             ztargets['NUMOBS_MORE'][ii] = 1
-
+    if survey == 'main':
+        # If the object is confirmed to be a tracer QSO, then don't request more observations
+        if (obsconditions.mask(obscon) & obsconditions.mask("DARK")) != 0:
+            if zcat is not None:
+                ii = ztargets['SPECTYPE']=='QSO'
+                ii &= (ztargets['ZWARN'] == 0 ) 
+                ii &= (ztargets['Z']<2.15)
+                ii &= (ztargets['NUMOBS']>0)
+                ztargets['NUMOBS_MORE'][ii] = 0
```

Comment:
@geordie666 (or others): what should we do for spectype=QSO, z<2.15, and zwarn>0?  As implemented in this PR that case would get more observations but I'm not sure that is what we want.  I thought the default policy was "only continue observing if definitively confirmed to be QSO z>2.15" rather than "only stop observing if definitively confirmed to be QSO z<2.15 (i.e. keep observing ambiguous cases too)"

I know the LyA group (@andreufont in particular) have floated the idea of getting a second observation for ambiguous cases, but stopping if a second observation doesn't return a definitive answer.  Implementing that might need a deeper code update.

If we know what the current default plan is, let's implement that here.  If we still need discussion to refine that plan, we could go ahead and merge this PR as a step in the right direction (since this prevents confirmed z<2.15 QSOs from being re-observed) and then make additional updates in future PRs.

Reply:
Hi @sbailey, we have been meaning to look at this closely for years, but none has managed to provide a definite answer.

Some issues we have faced so far:
 - When running tests of the classifiers on DESI mocks, the performance is too good, at least when compared to BOSS/eBOSS data. We believe this is due to the quasar templates in desisim do not have enough diversity.
 - We haven't been able to integrate the classifiers to desihub, because they depend on Scikit Learn or TensorFlow. For the rest of the discussion, we could assume that we will only use Redrock (since this is already integrated).
 - We still haven't figure out how to integrate the result of the different classifiers to the main pipeline (fiber re-assignement). I believe @geordie666 mentioned that he could be able to help here.
 - We still haven't been able to run our survey simulations, including fiber re-assignement, to be able to test the software I mention above. Julian Bautista tried for a while, but he got stuck early on in some issues with mini-test.
 
Given the difficulties described above, I'd really like to have a baseline that does not assume that we will solve all the issues above in the next couple of months (we will try tough!). It would be great if we could have the following baseline:
 - stop observing if definitively confirmed that it is NOT a QSO z > 2.1

I'd like to test in survey simulations how many fibers we would need to do that, although again, the tests will be as good as the quasar templates...

Reply:
Thanks for the comments @andreufont .  In main operations we will (hopefully) have multiple codes checking and more human oversight, but the same policy questions will still apply (e.g. there will still be some ambiguous maybe-QSOs that we'll have to decide what to do).  I agree that the default pipeline+redrock should at the policy level do the right thing, such that other codes improve the correctness / purity of the decisions, but don't fundamentally alter the policies.

> * stop observing if definitively confirmed that it is NOT a QSO z > 2.1

That is what is implemented here (almost, z 2.15 instead of 2.1), with the consequence that
 
* continue observing if definitively confirmed that it *is* QSO z>2.1 (good)
* continue observing if the redshift answer is ambiguous (maybe that's what we want, maybe not, especially if it is still ambiguous after the second observation).

@andreufont can you confirm what you think we should do in the case of an ambiguous redshift?  (but no promises that we'll implement that right now depending on how complex it is...)